lib/worker.js

```diff
@@ -84,30 +80,78 @@ async function getLocalLinter(basedir) {
             linter = require('loophole').allowUnsafeNewFunction(() => require(linterPath).Linter);
           }
         } else {
-          linter = tslintDef;
+          return resolve();
         }
-        tslintCache.set(basedir, linter);
+        tslintCache.set(fileDir, linter);
         return resolve(linter);
       },
     );
   });
 }
 
+function getNodePrefixPath() {
+  return new Promise((fulfill, reject) => {
```

Comment:
🐎 This is _extrememely_ unlikely to change during runtime, cache it after the first (successful) call and return that.

Reply:
❓ Any particular reason you used `fulfill` instead of the standard `resolve`? I wouldn't block over it, but the non-standard nature of it requires a double take on the code.

Reply:
The npm prefix doesn't actually have to be cached because this function is never called more than once. If it ever gets to the point that it needs to grab the prefix (because there was no local linter, and the user set the "use global tslint" option but didn't explicitly give their prefix in the settings) then it immediately uses the prefix to grab the global linter. If it's able to resolve it, then it caches that one as the "fallback linter," and if it's not able to resolve it, it caches linter-tslint's version as the "fallback linter." Either way, this function should never be called again, because `getLinter()` will never get past line 124 once `fallbackLinter` is set. 👍

As for the fulfill/resolve thing, I'm not actually sure... I thought maybe I had seen it in linter-eslint or something and copied the style, but now I can't find it anywhere. Dunno where I got it from, to be honest. xD I'll just change it to `resolve`, I don't really care either way and resolve does seem more "standard," as you said.

Reply:
Makes sense about the calling of `getNodePrefixPath`, thanks for clearing that up!

`resolve` is actually part of the standard documentation ([Promise](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise)), but you _can_ use whatever name you want... almost everyone just uses the standard names though lol.